src/v/kafka/server/handlers/produce.cc

```diff
@@ -354,9 +355,10 @@ produce_handler::handle(request_context ctx, ss::smp_service_group ssg) {
      * authorization failed.
      */
     if (request.has_transactional) {
-        // will be removed when all of transactions are implemented
-        return ctx.respond(request.make_error_response(
-          error_code::transactional_id_authorization_failed));
+        if (!config::shard_local_cfg().enable_transactions.value()) {
```

Comment:
maybe set a flag `ctx.server().enable_transactions` on the server to cache this config value?